File: Backend/Data/ViewModels/Types/CommentType.cs
namespace Backend.Data.ViewModels.Types
{
    /// <summary>
    /// Class that defines comment type
    /// </summary>
    public class CommentType
    {
        public string UserName { get; set; }
        public string Comment { get; set; }

        /// <summary>
        /// Class's contructor method
        /// </summary>
        /// <param name="userName">User name of user that commented the route</param>
        /// <param name="comment">User's commnent</param>
        public CommentType(string userName, string comment)
        {
            this.UserName = userName;
            this.Comment = comment;
        }

        public CommentType()
        {
            this.UserName = null!;
            this.Comment = null!;
        }
    }
}

File: Frontend/src/app/viewModels/Types/CommentType.ts

/**
 * This type represents a comment from a specific user.
 */
export class CommentType {

    /**
     * Construct's an instance of {@link CommentType}
     * 
     * @param userName user name of the user who made the comment.
     * @param comment comment made by the user.
     */
    constructor(
        public userName: string,
        public comment: string
    ) { }
}